WakeLockRequest is independent rather than WakeLock

Honry opened the pull request:
- Test wake lock state is global
  - Test two wake lock returns same object
  - Fixed #8411

<!-- Reviewable:start -->

<!-- Reviewable:end -->

w3c-bots commented:
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/21013)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/306621414)

Honry committed:
WakeLockRequest is independent rather than WakeLock

  - Test wake lock state is global
  - Test two wake lock returns same object
  - Fix #8411

Honry committed:
Fix traling whitespace

marcoscaceres commented on wake-lock/wakelock-state-is-global.https.html:
```diff
@@ -0,0 +1,74 @@
+
+<body>
+<script id="iframe" type="text/plain">
+let iframeWakeLock, iframeRequest;
```
Hmm... this is super racy... we can live with it... but not my favorite.

marcoscaceres commented on wake-lock/wakelock-state-is-global.https.html:
```diff
@@ -0,0 +1,74 @@
+  assert_true(isActive1, "the iframe wake lock state is actived when iframe wake lock is acquired");
+  assert_true(wakeLock.active, "the wake lock state is actived when iframe wake lock is acquired");
+
+
```
nit: remove this line.

marcoscaceres reviewed (changes requested):
I'm a bit concerned about `wakelock-state-is-global.https.html`, tbh... the design of using a global  is quite racy. I could be convinced to live with it, but can you maybe come up with a better way of keeping state than relying on globals?

marcoscaceres commented:
If you can't come up with anything better for wakelock-state-is-global.https.html, that's ok... let me know.

Honry commented:
@marcoscaceres, thanks for your review, I'm afraid I have no much better idea, @andrey-logvinov, do you have any good opinion?

Honry committed:
fix nit

marcoscaceres reviewed (approved):
I guess we have enough flow control in the test that should prevent race conditions. Hopefully this won't come back to bite us.

Honry commented:
@marcoscaceres, thanks, I hope so. :)

marcoscaceres commented:
Ok, merging to move forward, but @andrey-logvinov, please do take a look and file any bugs you might find (or if you can think of another way of testing this).

andrey-logvinov commented on wake-lock/wakelock-promise.https.html:
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>navigator.getWakeLock() for the same Document always return same WakeLock promise</title>
```
It is not required to always return the same promise - it is only required that if a returned promise is ever resolved or rejected, all returned promises must be resolved or rejected with the same value.

Honry replied:
> it is only required that if a returned promise is ever resolved, all returned promises must be resolved with the same value. 

About this checkpoint, isn't the test satisfied?  For sure, the desc is a bit misunderstanding.

> Also rejections must be consistent, e.g. if one promise is rejected with SecurityException, the other must be rejected also with SecurityException, but it is not required for two exceptions to be the same object.

But how to prove that the rejection is caused by other rejected promise in same Document? Since the rejected condition is global, they should be rejected beyond doubt.

andrey-logvinov replied:
Sorry for late response.
If I understand it correctly the test asserts that the same promise object is returned, which necessarily implies that the resolution is also the same. But the spec does not require to return exactly the same promise. This is not very relevant from the script point of view but it might complicate an implementation e.g. if it is technically easier to return different promise objects resolved with the same value than exactly the promise object. For now I think the test is adequate.

As for the rejection I agree that it is effectively untestable because it requires an anticipated scenario where the rejection might not be consistent but there is no such scenario as rejection conditions are constant in time for the same Document.
